sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -223,8 +223,6 @@ def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
                     break
                 except Exception as exception:  # pylint: disable=broad-except
                     self._amqp_transport.check_link_stolen(self, exception)
-                    if not self.running:  # exit by close
-                        return
```

Comment:
@annatisch - removing this results in raising the exception once the retries are exhausted. The event processor catches this except, processes the error, and closes the consumer + partition with "OwnershipLost" as the reason. It'll then re-initialize the partition and consumer and continue running.

If this is left in here, when retry_total is > 0, the error callback is never called b/c the error is never raised and caught by the event processor.

Reply:
That seems reasonable - though a couple of things to maybe investigate before we simply remove this:
- This is in a broad-except clause, could we better resolve this by using a finer-grained except clause? For example, perhaps if `close()` is called, it raises a particular error that we can `return` on?
- Why is `self.running` set to `False` in the case of the error scenario you describe? Perhaps we should only be explicitly settings this to `False` in the `close()` scenario?

Reply:
TL;DR - I've reverted this and created an issue (#27137) to resolve this after GA, since it seems to be a bug from before in previously GA'ed versions.

More details:
So running is being set to False b/c:
 1) The first time there's an exception here, consumer._handle_exception is called on line 230.
 2) In handle_exception, if the error is an EventHubError, we call [consumer.close_handler()](https://github.com/Azure/azure-sdk-for-python/blob/84f05915c7a7d66c0f11b03d8b290e337ee970ac/sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_uamqp_transport.py#L609)
3) Inside close_handler(), we close the consumer and set `self.running = False`.